VideoLocker/src/main/java/org/edx/mobile/view/CourseDiscussionPostsBaseFragment.java

```diff
@@ -68,10 +59,26 @@ public void onItemClick(AdapterView<?> parent, View view, int position, long id)
                     thread.setRead(true);
                     discussionPostsAdapter.getView(position, view, parent);
                 }
-
-                CourseDiscussionPostsBaseFragment.this.onItemClick(thread, parent, view,
-                        position, id);
             }
         });
     }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+
+        if (isRestart) {
+        /*
+        If the activity/fragment needs to be reinstantiated upon restoration,
+        then in some cases the onRestart() callback maybe invoked before view
+        initialization, and thus the controller might not be initialized, and
+        therefore we need to guard this with a null check.
+         */
+            if (controller != null) {
+                nextPage = 1;
+                controller.resetSilently();
+            }
+        }
+        isRestart = true;
```

Comment:
Logically we are on the lookout for a restart, which is easily readable/understandable from the current implementation.

Otherwise, the function will read like "if fragment `isStopped` then `do a silent refresh`" which isn't what we want to imply 😉